pkg/logql/engine.go

```diff
@@ -192,6 +193,12 @@ func (q *query) Exec(ctx context.Context) (logqlmodel.Result, error) {
 	log, ctx := spanlogger.New(ctx, "query.Exec")
 	defer log.Finish()
 
+	if GetRangeType(q.params) == InstantType {
```

Comment:
WDYT of adding a runtime config (like log_queries) and only logging this in case it is enabled? not sure how costly it will be to always log all queries

Reply:
The challenge here is we often don't know the tenant that is responsible for the OOM's so we'd need a way to enable it for everyone.

We already log metrics.go with a result for every query at info level so it seemed reasonable to me to log the start of the query also at info level.

I think this is an acceptable tradeoff, we can always revisit though.

Reply:
Ooh I like that idea, then we can log all query types

Reply:
@slim-bean what's the rationale around only enabling for instant queries? Surely we also see OOMs for large metric queries sometimes? This will definitely help in the case of rulers though, so hell yeah from that perspective

Reply:
Fair enough. I like that we're starting with level=info so that a new rollout isn't necessary in case you need to debug something. On the other hand, my concern with logging all queries is that we'll end up logging even subqueries AFAIK, so the volume will be huge I suppose.

Anyway, LGTM!

For the future, would be really cool if we could have a runtime configuration section that although loaded in runtime isn't per-tenant. Would be really helpful for scenarios similar to this one.